src/utils.c

```diff
@@ -209,21 +217,19 @@ void trim(char *str)
 
 void find_replace_extension(char *str, char *orig, char *rep, char *output)
 {
-    char* buffer = (char*)calloc(8192, sizeof(char));
-
-    sprintf(buffer, "%s", str);
+    char* buffer = strdup(str);
```

Comment:
Maybe there is the same problem as in the previous function?

Reply:
strdup allocates and copies a string,so p points to a heap string.See the man page for it .